keras_nlp/samplers/contrastive_sampler_test.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for Top-K sampler."""
+
+import tensorflow as tf
+from absl.testing import parameterized
+
+from keras_nlp.samplers.contrastive_sampler import ContrastiveSampler
+
+
+class ContrastiveSamplerTest(tf.test.TestCase, parameterized.TestCase):
+    def setUp(self):
+        super().setUp()
+        # Use a simple alphabet of lowercase characters to [0, 26).
+        self.int_lookup = {i: chr(i + ord("a")) for i in range(26)}
+        self.char_lookup = {v: k for k, v in self.int_lookup.items()}
+        self.batch_size = 1
+        self.length = 12
+        self.hidden_dim = 3
+        self.vocab_size = len(self.int_lookup)
+        self.hidden_states = tf.ones(
+            [
+                self.batch_size,
+                self.length,
+                self.hidden_dim,
+            ]
+        )
+
+        def next(prompt, cache, index):
+            batch_size = tf.shape(prompt)[0]
+            # Return a distribution favoring the next char in cache.
+            logits = tf.one_hot(cache[:, index], self.vocab_size) * 1e9
+            hidden_states = tf.ones([batch_size, 1, self.hidden_dim])
```

Comment:
If we remove the shape confusing here, this could be 

```
hidden_states = tf.expand_dims(cache[:, index], -1)
logits = tf.one_hot(cache[:, index], self.vocab_size) * 1e9
```

Which would be a little closer to a "real world" flow here.